modules/parsePath.js

```diff
@@ -1,16 +1,18 @@
 import warning from 'warning'
 
-function extractPath(string) {
+export function extractPath(string, fullUrl = false) {
   const match = string.match(/^https?:\/\/[^\/]*/)
 
   if (match == null)
     return string
 
-  warning(
-    false,
-    'A path must be pathname + search + hash only, not a fully qualified URL like "%s"',
-    string
-  )
+  if (!fullUrl) {
+    warning(
+      false,
```

Comment:
Just pass in `fullUrl` here instead of `false` instead of adding the explicit branch.